py/runtime.c

```diff
@@ -814,7 +814,7 @@ void mp_call_prepare_args_n_kw_var(bool have_self, size_t n_args_n_kw, const mp_
                     mp_obj_t iterable = mp_getiter(arg, &iter_buf);
                     mp_obj_t item;
                     while ((item = mp_iternext(iterable)) != MP_OBJ_STOP_ITERATION) {
-                        if (args2_len >= args2_alloc) {
+                        if (args2_len - (n_args - i) >= args2_alloc) {
```

Comment:
I think this should be `args2_len + (n_args - i)`, otherwise it can go negative.

I think the test only passes because when it goes negative it becomes a large positive number that does satisfy this test.

If you change the code here to `args2_len >= args2_alloc + (n_args - i)` then the test that's added in this PR fails.

Reply:
🤦 Thanks for catching that.

Reply:
fixed